Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/auditrepository/hibernate/AuditRepositoryDAO.java

```diff
@@ -161,4 +166,150 @@ public List queryAuditRepositoryOnCriteria(String eUserId, String ePatientId, Da
+
+            if (NullChecker.isNotNullish(messageId) || NullChecker.isNotNullish(relatesTo)) {
+
+                if (NullChecker.isNotNullish(messageId) && (messageId.startsWith("urn:uuid:")
+                    || !messageId.startsWith("urn:uuid:"))) {
+                    if (messageId.endsWith("%")) {
+                        queryCriteria.add(Restrictions.like("messageId", messageId));
+                    } else {
+                        queryCriteria.add(Restrictions.like("messageId", messageId + "%"));
+                    }
+
+                }
+
+                if (NullChecker.isNotNullish(relatesTo)) {
+                    if (relatesTo.endsWith("%")) {
+                        queryCriteria.add(Restrictions.like("relatesTo", relatesTo));
+
+                    } else {
+                        queryCriteria.add(Restrictions.like("relatesTo", relatesTo + "%"));
+
+                    }
+                }
+
+            } else if ((outcome != null && outcome >= 0) || NullChecker.isNotNullish(eventTypeList)
+                || NullChecker.isNotNullish(userId) || NullChecker.isNotNullish(remoteHcidList)
+                || NullChecker.isNotNullish(startDate) || NullChecker.isNotNullish(endDate)) {
+
+                if (outcome != null && outcome >= 0) {
+                    queryCriteria.add(Restrictions.eq("outcome", outcome));
+                }
+
+                if (NullChecker.isNotNullish(eventTypeList)) {
+                    queryCriteria.add(Restrictions.in("eventType", eventTypeList));
+                }
+
+                if (NullChecker.isNotNullish(userId)) {
+                    queryCriteria.add(Restrictions.eq("userId", userId));
+                }
+
+                if (NullChecker.isNotNullish(remoteHcidList)) {
+                    queryCriteria.add(Restrictions.in("remoteHcid", remoteHcidList));
+                }
+
+                if (NullChecker.isNotNullish(startDate) && NullChecker.isNotNullish(endDate)) {
+                    SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                    queryCriteria.add(Restrictions.between("eventTimeStamp", new Date(sdf.parse(startDate).getTime()),
+                        new Date(sdf.parse(endDate).getTime())));
+
+                } else if (NullChecker.isNotNullish(startDate) && NullChecker.isNullish(endDate)) {
+                    SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+                    queryCriteria.add(Restrictions.ge("eventTimeStamp", new Date(sdf.parse(startDate).getTime())));
+
+                }
+            }
+
+            queryCriteria.setProjection(Projections.projectionList()
+                .add(Projections.property("userId"))
+                .add(Projections.property("eventType"))
+                .add(Projections.property("eventId"))
+                .add(Projections.property("outcome"))
+                .add(Projections.property("eventTimeStamp"))
+                .add(Projections.property("remoteHcid"))
```

Comment:
eventTimeStamp should be eventTimestamp. MySQL is case-insensitive for table and column name however it depends on the operating system as well. So your code probably works fine on windows but will have issues on Linux/unix machines.